doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -556,87 +546,72 @@ the conversion in hypotheses :n:`{+ @ident}`.
  The :math:`\delta` rule only applies to transparent constants (see :ref:`vernac-controlling-the-reduction-strategies`
  on transparency and opacity).
 
-.. tacn:: cbn
-          simpl
-   :name: cbn; simpl
+.. tacn:: simpl {? @delta_flag } {? {| @reference_occs | @pattern_occs } } {? @occurrences }
 
-   These tactics apply to any goal. They try to reduce a term to
-   something still readable instead of fully normalizing it. They perform
+   Reduce a term to
+   something still readable instead of fully normalizing it. It performs
    a sort of strong normalization with two key differences:
 
-   + They unfold a constant if and only if it leads to a :math:`\iota`-reduction,
+   + It unfolds a constant if and only if it leads to a :math:`\iota`-reduction,
      i.e. reducing a match or unfolding a fixpoint.
-   + While reducing a constant unfolding to (co)fixpoints, the tactics
-     use the name of the constant the (co)fixpoint comes from instead of
+   + While reducing a constant unfolding to (co)fixpoints, the tactic
+     uses the name of the constant the (co)fixpoint comes from instead of
      the (co)fixpoint definition in recursive calls.
 
-   The :tacn:`cbn` tactic was intended to be a more principled, faster and more
-   predictable replacement for :tacn:`simpl`.
+   Only transparent constants whose name can be reused in
+   recursive calls may be unfolded by :tacn:`simpl`. For instance, a
+   constant :g:`plus' := plus` may be unfolded and reused in
+   recursive calls, but a constant such as :g:`succ := plus (S O)` is
+   never unfolded.
 
-   The :tacn:`cbn` tactic accepts the same flags as :tacn:`cbv` and
-   :tacn:`lazy`. The behavior of both :tacn:`simpl` and :tacn:`cbn`
-   can be tuned using the :cmd:`Arguments` command.
+   :n:`{| @reference_occs | @pattern_occs }` can limit the application of :tacn:`simpl`
+   to:
 
-   .. todo add "See <subsection about controlling the behavior of reduction strategies>"
-      to TBA section
+   - subterms matching a pattern :n:`@term`
+   - the applicative subterms whose head occurrence is the unfoldable
+     constant :n:`@qualid`
+   - the applicative subterms whose head occurrence is the constant used
+     in the notation :n:`@string`
```

Comment:
The `at` clause should also be explained. Also, the grammar of `@reference_occs` and `@pattern_occs` should be moved here so that their documentation is easy to find (from other tactics that use them in their syntax).

Reply:
`@pattern_occs` is required if you give an `at`?  If you don't have a pattern to match, how does the `at` make sense?

There are 4-5 tactics that use `@reference_occs` and `@pattern_occs` directly, plus more through `@red_flags`.  Why move them here?

Reply:
> `@pattern_occs` is required if you give an at? If you don't have a pattern to match, how does the at make sense?

You can also use `@reference_occs` instead, in which case the pattern is morally the reference followed by any number of underscores.  Indeed "at" makes no sense for these tactics without a pattern/reference, (unlike `rewrite` and `change`, which I believe share the `@occurences` grammar entry and have an implied pattern) and hence we get an error message about not supporting occurrences when "at" shows up without a pattern.

Reply:
> There are 4-5 tactics that use `@reference_occs` and `@pattern_occs` directly, plus more through `@red_flags`. Why move them here?

I'm just saying that the grammar of `@reference_occs` and `@pattern_occs` should be in close vicinity to the explanation of what they do.